tests/test_config.py

```diff
@@ -102,26 +100,53 @@ def test_load_config_from_cli():
     assert config_obj.worker_concurrency == 5
 
 
-def test_load_config():
+def test_load_config(monkeypatch, tmp_path):
     # CLI config should override environment config
     cli = Namespace()
     cli.worker_concurrency = "5"
     # Environment config should override file config
-    os.environ["WORKER_CONCURRENCY"] = "15"
-    os.environ["AMQP_HOSTNAME"] = "localhost"
-    with NamedTemporaryFile("w") as f:
-        os.environ["VUMI_CONFIG_FILE"] = f.name
-        f.write(
-            """
-            amqp:
-                hostname: overwritten
-                port: 1234
-            worker_concurrency: 10
-            """
-        )
-        f.flush()
-        config = load_config(cli=cli)
+    monkeypatch.setenv("WORKER_CONCURRENCY", "15")
+    monkeypatch.setenv("AMQP_HOSTNAME", "localhost")
+
+    config_path = tmp_path / "test-config.yaml"
```

Comment:
What is this doing?

Reply:
`tmp_path` is a `pathlib.Path` object, so `tmp_path / "filename.ext"` is equivalent to `tmp_path.joinpath("filename.ext")`, which in turn is equivalent to `os.path.join(str_path, "filename.ext")` when working with strings instead of `Path`s.

Overriding the division operator to do path concatenation is a really neat readability trick once you get used to it. 🙂

Reply:
Yes, that is pretty cool thanks! Never seen it before